### PR TITLE
Expose move to slide method on slidescroll to enable flim-strip experience

### PR DESCRIPTION
**Pull request by camelburrito**
This is mainly done to enable (commerce) galleries to get something working. 
Will has a more comprehensive bind pr (https://github.com/ampproject/amphtml/pull/6797) that does a lot of awesome 2 way syncing.

### CODE REVIEW

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 1]
```diff
@@ -159,6 +159,17 @@ export class AmpSlideScroll extends BaseSlides {
...
           'touchend', this.touchEndHandler_.bind(this));
     }
+
+    this.registerAction('moveToSlideIndex', invocation => {
+      const args = invocation.args;
+      if (!args) {
+        return;
+      }
+      let slideIndex = Number(args['newIndex']);
+      if (!isNaN(slideIndex)) {
```
Comment (choumx) [inline extensions/amp-carousel/0.1/slidescroll.js]: `isFinite`?

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 2]
```diff
@@ -159,6 +159,17 @@ export class AmpSlideScroll extends BaseSlides {
       this.slidesContainer_.addEventListener(
           'touchend', this.touchEndHandler_.bind(this));
     }
+
+    this.registerAction('moveToSlideIndex', invocation => {
```
Comment (choumx) [inline extensions/amp-carousel/0.1/slidescroll.js]: Nit: How about `changeSlide`? Either way, probably should be consistent with the new event in #6797.

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 3]
```diff
@@ -159,6 +159,17 @@ export class AmpSlideScroll extends BaseSlides {
...
+      const args = invocation.args;
+      if (!args) {
+        return;
+      }
+      let slideIndex = Number(args['newIndex']);
```
Comment (choumx) [inline extensions/amp-carousel/0.1/slidescroll.js]: Nit: No need to disambiguate "index" with "new". How about just `'slide'`?

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 2 above)
Comment (mkhatib) [inline extensions/amp-carousel/0.1/slidescroll.js]: or `goTo`, `goToSlide`

Comment (camelburrito) [inline extensions/amp-carousel/0.1/slidescroll.js]: @choumx - i would want it to be consistent with yours, also i want it to be explicit and say include the word `index` let me know if you can comeup with something for me :)

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 3 above)
Comment (camelburrito) [inline extensions/amp-carousel/0.1/slidescroll.js]: trying to be consistent with rest of the file (naming it slide can be super confusing in the slide.)

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 2 above)
Comment (choumx) [inline extensions/amp-carousel/0.1/slidescroll.js]: IMO, `slide` is a more specific form of `index` and is closer to colloquial usage. E.g. when presenting a slide deck, one would say "go to the next slide". 

`goToSlide` works in this form but "on: go to slide" doesn't read naturally. For example:

`<amp-carousel type=slides on="goToSlide:otherCarousel.goToSlide(index=3)">`

vs.

`<amp-carousel type=slides on="changeSlide:otherCarousel.changeSlide(slide=3)">`

This is nitpicky though and I don't really mind. Feel free to get a PM to tiebreak or ACK this comment. :)

Comment (camelburrito) [inline extensions/amp-carousel/0.1/slidescroll.js]: amp-carousel does not have any on: events that are supported - i like your first option.

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 1 above)
Comment (camelburrito) [inline extensions/amp-carousel/0.1/slidescroll.js]: Done

**Comment by camelburrito** [comment]
Ptal

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 4]
```diff
@@ -159,6 +159,17 @@ export class AmpSlideScroll extends BaseSlides {
...
+
+    this.registerAction('goToSlide', invocation => {
+      const args = invocation.args;
+      if (!args) {
+        return;
+      }
+      const newIndex = Number(args['index']);
+      if (isFinite(newIndex)) {
+        this.showSlide_(newIndex);
+      }
```
Comment (choumx) [inline extensions/amp-carousel/0.1/slidescroll.js]: Else: warn or error?

Comment (camelburrito) [inline extensions/amp-carousel/0.1/slidescroll.js]: Done

**Pull request merged**